.github/actions/run_local_pytest/action.yml

```diff
@@ -0,0 +1,48 @@
+name: 'Install Local Python dependencies'
+description: 'Install Local Python dependencies'
+inputs:
+  job_type:          {default: '', type: string, description: Specifies which steps should be executed}
+  build_type:        {default: 'release', type: string, description: Specifies whether the build that we are testing is release or debug}
+  threads:           {default: '1', type: string, description: Specifies the number of threads that will be used for tests }
+  fast_tests_only:   {default: '0', type: string, description: Specifies whether we want to run only the fast tests or not }
+  other_params:      {default: '', type: string, description: Other params that should be passed to the python command }
+runs:
+  using: "composite"
+  steps:
+    - name: Install local dependencies with pip
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+
+        python >build_tooling/requirements-analysis-workflow.txt <<END
+        from configparser import ConfigParser
+        cf = ConfigParser()
+        cf.read("setup.cfg");
+
+        # setup_requires is not needed
+        print(cf['options']['setup_requires'])
+        print(cf['options']['install_requires'])
+        print(cf['options.extras_require']['Testing'])
+        END
+
+        python -m pip install -r build_tooling/requirements-analysis-workflow.txt
+        python setup.py protoc --build-lib python
+
+    - name: Run local pytests
+      if: inputs.job_type == 'run'
+      shell: bash
+      run: |
+        yum install nodejs npm -y
+        npm config set strict-ssl false
+        npm install -g azurite
+        
+        cd python
+        ln -s ../cpp/out/linux-${{ inputs.build_type }}-build/arcticdb/arcticdb_ext.cpython-36m-x86_64-linux-gnu.so
+        export ARCTICDB_RAND_SEED=$RANDOM
+        python ${{inputs.other_params}}-m pytest -n ${{ inputs.threads }} tests
```

Comment:
Minor: the shell doesn't mind if you use multiple spaces to separate arguments, so you could have a space before the `-m`